components/Elevation/src/UIView+MaterialElevationResponding.m

```diff
@@ -0,0 +1,84 @@
+// Copyright 2019-present the Material Components for iOS authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "UIView+MaterialElevationResponding.h"
+
+#import "MDCElevation.h"
+#import "MDCElevationOverride.h"
+
+@implementation UIView (MaterialElevationResponding)
+
+- (CGFloat)mdc_baseElevation {
+  CGFloat totalElevation = 0;
+  UIView *current = self;
+
+  while (current != nil) {
+    id<MDCElevation> elevatableCurrent = [current objectConformingToElevationInResponderChain];
+    if (current != self) {
+      totalElevation += elevatableCurrent.mdc_currentElevation;
+    }
+    id<MDCElevationOverride> elevatableCurrentOverride =
+        [current objectConfromingToOverrideInResponderChain];
+    if (elevatableCurrentOverride != nil) {
```

Comment:
i think this check can be removed

Reply:
We do not, we shouldn't be checking for nil by default because it's okay to call a method on nil but since we need the break in the other case we need to check.

Reply:
Updated to not be nested.